Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -941,6 +941,11 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
     case AtomicsIsLockFree: {
         if (m_graph.child(node, 0).useKind() != Int32Use)
             clobberWorld();
+        JSValue value = m_state.forNode(m_graph.child(node, 0)).value();
+        if (value.isNumber()) {
+            node->clearFlags(NodeMustGenerate);
+            m_state.setShouldTryConstantFolding(true);
+        }
```

Comment:
I think this should be 
```
else {
    node->clearFlags(NodeMustGenerate);
    if (JSValue value = m_state.forNode(m_graph.child(node, 0)).value())
        m_state.setShouldTryConstantFolding(true);
}
```
We want to allow the node to be cleared if it has no side effects even if we can't constant fold it. We also need to add a constant folding rule since I don't see one. Or you could just not add the constant folding part.